packages/medusa/src/strategies/product-export.ts

```diff
@@ -0,0 +1,781 @@
+           moneyAmountShape;
+      `)
+
+    this.appendOptionsDescriptors(maxOptionsCount)
+    this.appendMoneyAmountDescriptors(maxMoneyAmountCount)
+    this.appendImagesDescriptors(maxImagesCount)
+
+    return [...this.columnDescriptors.keys(), this.NEWLINE].join(
+      this.DELIMITER_
+    )
+  }
+
+  private appendImagesDescriptors(maxImagesCount: number): void {
+    for (let i = 0; i < maxImagesCount; ++i) {
+      this.columnDescriptors.set(`Image ${i + 1} Url`, {
+        accessor: (image: Image) => image?.url ?? "",
+        entityName: "image",
+        index: i,
+      })
+    }
+  }
+
+  private appendOptionsDescriptors(maxOptionsCount: number): void {
+    for (let i = 0; i < maxOptionsCount; ++i) {
+      this.columnDescriptors
+        .set(`Option ${i + 1} Name`, {
+          accessor: (productOption: ProductOption) =>
+            productOption?.title ?? "",
+          entityName: "option",
+          index: i,
+        })
+        .set(`Option ${i + 1} Value`, {
+          accessor: (variantOption: ProductVariantOption) =>
+            variantOption?.value,
+          entityName: "variantOption",
+          index: i,
+        })
+    }
+  }
+
+  private appendMoneyAmountDescriptors(maxMoneyAmountCount: number): void {
+    for (let i = 0; i < maxMoneyAmountCount; ++i) {
+      this.columnDescriptors
+        .set(`Price ${i + 1} Currency code`, {
+          accessor: (moneyAmount: MoneyAmount) =>
+            moneyAmount?.currency_code?.toLowerCase() ?? "",
+          entityName: "moneyAmount",
+          index: i,
+        })
+        .set(`Price ${i + 1} Region name`, {
+          accessor: (moneyAmount: MoneyAmount) =>
+            moneyAmount?.region?.name?.toLowerCase() ?? "",
+          entityName: "moneyAmount",
+          index: i,
+        })
+        .set(`Price ${i + 1} Amount`, {
+          accessor: (moneyAmount: MoneyAmount) =>
+            moneyAmount?.amount?.toString() ?? "",
+          entityName: "moneyAmount",
+          index: i,
+        })
+    }
+  }
+
+  private buildProductLines(product: Product): string[] {
+    const outputLineData: string[] = []
+
+    const productLine = this.buildProductLineData(product)
+    outputLineData.push(productLine)
+
+    const variantLines = this.buildProductVariantsLineData(
+      product,
+      product.variants
+    )
+    outputLineData.push(...variantLines)
+
+    return outputLineData
+  }
+
+  private buildProductLineData(product: Product): string {
+    const productLineData: string[] = []
+    for (const [, columnSchema] of this.columnDescriptors.entries()) {
+      if (columnSchema.entityName === "product") {
+        productLineData.push(columnSchema.accessor(product))
+        continue
+      }
+      if (
+        columnSchema.entityName === "option" &&
+        columnSchema.index !== undefined
+      ) {
+        const option = product?.options[columnSchema.index] ?? {}
+        productLineData.push(columnSchema.accessor(option))
+        continue
```

Comment:
suggestion: abstract the accessor with an optional index value used in these methods. 

This would prevent the need for an individual "if" statement for each of the relations.